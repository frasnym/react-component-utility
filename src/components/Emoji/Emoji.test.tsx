import React from 'react'
import renderer from 'react-test-renderer'
import { render, screen } from '@testing-library/react'
import { Emoji } from '.'

describe('Emoji', () => {
  it('should renders the Emoji component', async () => {
    const tree = renderer.create(<Emoji symbol="ðŸ¤©" />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should renders the Emoji component without label', async () => {
    render(<Emoji symbol="ðŸ¤©" />)

    const span = await screen.findByText('ðŸ¤©')
    expect(span.getAttribute('role')).toEqual('img')
    expect(span.getAttribute('aria-label')).toEqual('')
    expect(span.getAttribute('aria-hidden')).toEqual('true')
  })

  it('should renders the Emoji component with label', async () => {
    const label = '[TEST] label'
    render(<Emoji symbol="ðŸ¤©" label={label} />)

    const span = await screen.findByText('ðŸ¤©')
    expect(span.getAttribute('role')).toEqual('img')
    expect(span.getAttribute('aria-label')).toEqual(label)
    expect(span.getAttribute('aria-hidden')).toEqual('false')
  })
})
