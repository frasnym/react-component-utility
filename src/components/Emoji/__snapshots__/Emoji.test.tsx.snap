// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Emoji should renders the Emoji component 1`] = `
<span
  aria-hidden="true"
  aria-label=""
  role="img"
>
  ðŸ¤©
</span>
`;
